packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -91,15 +99,17 @@ export class CrossChainMessenger implements ICrossChainMessenger {
           message.target,
           message.message,
           overrides?.l2GasLimit ||
-            (await this.provider.estimateL2MessageGasLimit(message))
+            (await this.provider.estimateL2MessageGasLimit(message)),
+          omit(overrides || {}, 'l2GasLimit')
         )
       } else {
```

Comment:
This `else` is redundant since the `if` branch returns

Reply:
Ehhh I prefer the explicit if/else, even if it's redundant.